### PR TITLE
test: fix race in check-docker

### LINKED ISSUE #2696: Race in check-docker
The check-docker test still races, even after the fix in 5880737829eecce62ccbcfabf3a95fa7ac468b31. Although it races a lot more rarely.

The race is between the echo and the nc here:

```
CMD ["/bin/sh", "-c", "echo '%s' | tee /dev/fd/2 | nc -l -p %d"]
```

We need to be able to echo something after nc has its port open and is listening.

### PR DESCRIPTION
**Pull request by dperpeet**
Wait for netcat to listen on port before we echo a terminal message.

Fixes #2696

### CODE REVIEW

File: test/check-docker [hunk 1]
```diff
@@ -176,14 +176,33 @@ class TestDocker(MachineCase):
         port = 3380;
         message = "HelloMessage."
 
+        # background: wait for the port to become available and echo message
+        # then start netcat server for port
+        # netcat has to run in the main process, otherwise docker might kill the container prematurely
+        # busybox netcat has reduced functionality (no --send-only or -c parameters)
+        # after 10 seconds without a port connection, the message "port not open" is printed instead
+        def echoCommand(listen_port, message_to_send):
```
Comment (stefwalter) [inline test/check-docker]: Hmmm, this looks like it'll do the trick. But could we just hard code these and not make this a big substitution party?

Comment (stefwalter) [inline test/check-docker]: Or rather, could we just write this as a command into the the Docker image?

Comment (dperpeet) [inline test/check-docker]: We need this for different messages, different ports and in the second check, we explicitly test setting the run command in the run container dialog.

Is it worth uploading a file into the vm to add a script?

Comment (stefwalter) [inline test/check-docker]: I think it is worth uploading. We do it in the kubernetes tests.

**Pull request closed without merging**